### ISSUE TITLE
"Directives" field of @Template annotation should have optional checked-mode enforcement

### ISSUE THREAD

**Issue by matanlurey** [angular/angular#1138]
Assume I have a component, `Foo`, which is represented by:

```dart
@Component(selector: 'foo')
class FooComponent {}
```

In my parent component, `FooList`, I forget to pass `directives`:

```dart
@Component(selector: 'foo-list')
@Template(inline: '''
  <foo></foo>
  <foo></foo>
  <foo></foo>
''')
class FooListComponent {}
```

It would be very helpful, in development mode, if we could receive warnings when/if unregistered selectors are used in the template. In Dart, this could be done something like:

```dart
void checkIfAllSelectorsExist(
    Iterable<String> selectorsInTemplate, 
    List<String> selectorsRegistered) {
  assert(selectorsInTemplate.every(selectorsRegistered.contains));
}
```

(I assume something similar is available in the TS/Transpilation layer)

**Comment by yjbanov** [comment]
How would you tell an Angular component from a Web Component? That's the biggest question. I think we should be able to tell at runtime that an element is none of the following:

* a standard HTML element, e.g. `div`
* an Angular directive
* a web component

Then we could log a warning. But it's harder to do statically, unless the app developer also provides a whitelist of web components they use in their app.

**Comment by matanlurey** [comment]
I'd be OK with auto-magic for standard elements for now. This optional mode should only be used if the web components are whitelisted.

**Comment by naomiblack** [comment]
dup of #2014

**Comment by matanlurey** [comment]
Naomi, I don't see anything in #2014 that validates tag names.

**Comment by mhevery** [comment]
@matanlurey #2014 says that we need a schema. So if you say `<div ng-bar=foo>` and `ng-bar` is not in schema it will complain. 

I think this should be closed as a dup of #2014 Could you explain which use case is not covered?

**Comment by mhevery** [comment]
Closing as dup of #2014
